### PR TITLE
Sebastian/add calculator test

### PR DESCRIPTION
**Pull request by seblapela**
* #32

### CODE REVIEW

File: source/databricks/tests/integration/calculator/test_calculator.py [hunk 1]
```diff
@@ -85,3 +86,94 @@ def test_calculator_job_accepts_parameters_from_process_manager(
...
+        "2022-09-01T22:00:00Z",
+    ]
+
+    # Act
+    exit_code = subprocess.call(python_parameters)
+
+    result = spark.read.json(f"{delta_lake_path}/result")
+
+    # Assert
+    assert result.count() >= 100, "Calculator job failed to write files"
```
Comment (lasrinnil) [inline source/databricks/tests/integration/calculator/test_calculator.py:134]: maybe we could assert that two files are created, one for each grid area

File: source/databricks/tests/integration/calculator/test_calculator.py [hunk 2]
```diff
@@ -85,3 +86,94 @@ def test_calculator_job_accepts_parameters_from_process_manager(
...
+        "2022-05-31T22:00:00Z",
+        "--batch-period-end-datetime",
+        "2022-09-01T22:00:00Z",
+    ]
+
+    # Act
+    completed_process = subprocess.run(python_parameters, capture_output=True)
+
+    # Assert
+    assert "grid area" in completed_process.stderr.decode()
```
Comment (lasrinnil) [inline source/databricks/tests/integration/calculator/test_calculator.py:179]: maybe use the whole string from the raised exception

File: source/databricks/tests/integration/calculator/test_calculator.py [hunk 3]
```diff
@@ -85,3 +86,94 @@ def test_calculator_job_accepts_parameters_from_process_manager(
...
+    spark.read.json(f"{json_test_files}/integration_events.json").withColumn(
+        "body", col("body").cast("binary")
+    ).write.mode("overwrite").parquet(
+        f"{delta_lake_path}/parquet_test_files/integration_events"
+    )
+    spark.read.json(f"{json_test_files}/time_series_points.json").write.mode(
+        "overwrite"
+    ).parquet(f"{delta_lake_path}/parquet_test_files/time_series_points")
+    # Arrange
+    python_parameters = [
```
Comment (lasrinnil) [inline source/databricks/tests/integration/calculator/test_calculator.py:103]: move to fixture

File: source/databricks/package/balance_fixing_total_production.py [hunk 4]
```diff
@@ -239,23 +260,23 @@ def _get_result_df(enriched_time_series_points_df, batch_grid_areas) -> DataFram
         .withColumn(
             "quarter_times",
             when(
-                col("Resolution") == Resolution.hour,
+                col("Resolution") == repr(Resolution.hour),
```
Comment (johevemi) [inline source/databricks/package/balance_fixing_total_production.py:263]: You can write Resolution.hour.value to get the enum number value

File: source/databricks/package/balance_fixing_total_production.py [hunk 5]
```diff
@@ -218,8 +232,9 @@ def _get_enriched_time_series_points_df(
     enriched_time_series_point_df = timeseries_df.join(
         metering_point_period_df,
         (metering_point_period_df["GsrnNumber"] == timeseries_df["GsrnNumber"])
-        & (timeseries_df["time"] >= metering_point_period_df["EffectiveDate"])
-        & (timeseries_df["time"] < metering_point_period_df["toEffectiveDate"]),
+        # & (timeseries_df["time"] >= metering_point_period_df["EffectiveDate"])
+        # & (timeseries_df["time"] < metering_point_period_df["toEffectiveDate"])
```
Comment (johevemi) [inline source/databricks/package/balance_fixing_total_production.py:236]: Why is the commented out?

File: source/databricks/package/balance_fixing_total_production.py [hunk 6]
```diff
@@ -112,8 +120,9 @@ def _get_metering_point_periods_df(
     period_end_datetime,
 ) -> DataFrame:
     metering_point_events_df = (
-        raw_integration_events_df.withColumn("body", col("body").cast("string"))
-        .withColumn("body", from_json(col("body"), metering_point_generic_event_schema))
+        raw_integration_events_df.withColumn(
+            "body", from_json(col("body"), metering_point_generic_event_schema)
+        )
```
Comment (johevemi) [inline source/databricks/package/balance_fixing_total_production.py:125]: Is what is in body column already a string?

Comment (johevemi) [reply source/databricks/package/balance_fixing_total_production.py:125]: It get casted as string in line 59

**Comment by seblapela** [comment]
Test of stored time will be implemented in other test.

**Review by johevemi — APPROVED** [review]

**Pull request merged**